projects/OG-Analytics/src/test/java/com/opengamma/analytics/financial/interestrate/future/provider/InterestRateFutureOptionMarginNormalSmileMethodTest.java

```diff
@@ -172,4 +179,203 @@ public void presentValueNormalSensitivity() {
+      .createNormalSurfaceFuturesPricesSimpleMoneyness();
+  private static final NormalSTIRFuturesExpSimpleMoneynessProviderDiscount NORMAL_MULTICURVES_MONEYNESS = new NormalSTIRFuturesExpSimpleMoneynessProviderDiscount(
+      MULTICURVES, NORMAL_PARAMETERS_MONEYNESS, EURIBOR3M);
+
+  /**
+   * Test the option price from the future price.
+   */
+  @Test
+  public void priceMoneyness() {
+    double expiry = OPTION_ERU2.getExpirationTime();
+    EuropeanVanillaOption option = new EuropeanVanillaOption(STRIKE, expiry, IS_CALL);
+    double priceFuture = METHOD_FUTURES.price(ERU2, MULTICURVES);
+    double volatility = NORMAL_PARAMETERS_MONEYNESS.getZValue(expiry, STRIKE);
+    NormalFunctionData dataNormal = new NormalFunctionData(priceFuture, 1.0, volatility);
+    double priceExpected = NORMAL_FUNCTION.getPriceFunction(option).evaluate(dataNormal);
+    double priceComputed = METHOD_SECURITY_OPTION_NORMAL.price(OPTION_ERU2, NORMAL_MULTICURVES_MONEYNESS);
+    assertEquals("Future option with volatilities: option security price", priceExpected, priceComputed,
+        TOLERANCE_PRICE);
+  }
+
+  /**
+   * Test the option transaction present value.
+   */
+  @Test
+  public void presentValueMoneyness() {
+    double priceOption = METHOD_SECURITY_OPTION_NORMAL.price(OPTION_ERU2, NORMAL_MULTICURVES_MONEYNESS);
+    double presentValue1Expected = (priceOption - MARGIN_PRICE) * QUANTITY * NOTIONAL * FUTURE_FACTOR;
+    MultipleCurrencyAmount presentValue1Computed = METHOD_TRANSACTION_OPTION_NORMAL.presentValue(TRANSACTION_1,
+        NORMAL_MULTICURVES_MONEYNESS);
+    assertEquals("Future option with volatilities: option transaction pv", presentValue1Expected,
+        presentValue1Computed.getAmount(EUR), TOLERANCE_PV);
+    double presentValue2Expected = (priceOption - TRADE_PRICE) * QUANTITY * NOTIONAL * FUTURE_FACTOR;
+    MultipleCurrencyAmount presentValue2Computed = METHOD_TRANSACTION_OPTION_NORMAL.presentValue(TRANSACTION_2,
+        NORMAL_MULTICURVES_MONEYNESS);
+    assertEquals("Future option with volatilities: option transaction pv", presentValue2Expected,
+        presentValue2Computed.getAmount(EUR), TOLERANCE_PV);
+    MultipleCurrencyAmount presentValue1Calculator = TRANSACTION_1.accept(PVNFC, NORMAL_MULTICURVES_MONEYNESS);
+    assertEquals("Future option with volatilities: option transaction pv", presentValue1Computed.getAmount(EUR),
+        presentValue1Calculator.getAmount(EUR), TOLERANCE_PV);
+  }
+
+  /**
+   * Tests present value curve sensitivity.
+   */
+  @Test
+  public void presentValueCurveSensitivityMoneyness() {
+     MultipleCurrencyParameterSensitivity pvpsDepositExact = PSNFC.calculateSensitivity(TRANSACTION_1,
+        NORMAL_MULTICURVES_MONEYNESS, NORMAL_MULTICURVES_MONEYNESS.getMulticurveProvider().getAllNames());
+     MultipleCurrencyParameterSensitivity pvpsDepositFD = PSNFC_FD.calculateSensitivity(TRANSACTION_1,
```

Comment:
The Finite Difference code seems to create an incorrect data set (see comments on that part of the code). When more volatilities are added (see priceMoneyness test above), I expect this test to fail.